test/appsignal/oban_test.exs

```diff
@@ -12,34 +21,34 @@ defmodule Appsignal.ObanTest do
     assert attached?([:oban, :engine, :insert_job, :exception])
   end
 
-  test "does not attach to Oban events when :instrument_oban is set to false" do
-    :telemetry.detach({Appsignal.Oban, [:oban, :job, :start]})
-    :telemetry.detach({Appsignal.Oban, [:oban, :job, :stop]})
-    :telemetry.detach({Appsignal.Oban, [:oban, :job, :exception]})
-    :telemetry.detach({Appsignal.Oban, [:oban, :engine, :insert_job, :start]})
-    :telemetry.detach({Appsignal.Oban, [:oban, :engine, :insert_job, :stop]})
-    :telemetry.detach({Appsignal.Oban, [:oban, :engine, :insert_job, :exception]})
+  describe "when :instrument_oban is set to false" do
+    setup do
+      :telemetry.detach({Appsignal.Oban, [:oban, :job, :start]})
+      :telemetry.detach({Appsignal.Oban, [:oban, :job, :stop]})
+      :telemetry.detach({Appsignal.Oban, [:oban, :job, :exception]})
+      :telemetry.detach({Appsignal.Oban, [:oban, :engine, :insert_job, :start]})
+      :telemetry.detach({Appsignal.Oban, [:oban, :engine, :insert_job, :stop]})
+      :telemetry.detach({Appsignal.Oban, [:oban, :engine, :insert_job, :exception]})
 
-    with_config(%{instrument_oban: false}, fn -> Appsignal.start([], []) end)
+      with_config(%{instrument_oban: false}, fn -> Appsignal.start([], []) end)
 
-    assert !attached?([:oban, :job, :start])
-    assert !attached?([:oban, :job, :stop])
-    assert !attached?([:oban, :job, :exception])
-    assert !attached?([:oban, :engine, :insert_job, :start])
-    assert !attached?([:oban, :engine, :insert_job, :stop])
-    assert !attached?([:oban, :engine, :insert_job, :exception])
+      on_exit(fn ->
+        [:ok, :ok, :ok, :ok, :ok, :ok] = Appsignal.Oban.attach()
+      end)
+    end
 
-    [:ok, :ok, :ok, :ok, :ok, :ok] = Appsignal.Oban.attach()
+    test "does not attach to Oban events" do
+      assert !attached?([:oban, :job, :start])
+      assert !attached?([:oban, :job, :stop])
+      assert !attached?([:oban, :job, :exception])
+      assert !attached?([:oban, :engine, :insert_job, :start])
+      assert !attached?([:oban, :engine, :insert_job, :stop])
+      assert !attached?([:oban, :engine, :insert_job, :exception])
+    end
   end
```

Comment:
I added similar tests for Finch and Ecto when adding this one; if this one was causing problems, you may want to check those out as well.